text/0240-app-telemetry-schema-vision-roadmap.md

```diff
@@ -0,0 +1,384 @@
+indicative of a lack of standardized mechanisms for extending, customizing, and
+developing an ecosystem of schemas and semantic convention registries.**
+
+In response to these problems, a hierarchy of telemetry schemas can be defined,
+ranging from the most general to one that is specifically refined for an
+application. Each child schema inherits the properties of the parent schema and
+can, if necessary, override these properties locally. Additionally, any
+telemetry schema can import one or several semantic convention registries,
+allowing for the definition of OpenTelemetry, vendor-specific, and
+enterprise-level registries. These two enhancements make OpenTelemetry more
+customizable, extensible, and ecosystem-friendly.
+
+The following section will elaborate on the concepts of the Component Telemetry
+Schema, Resolved Telemetry Schema, Semantic Convention Registries, and their
+relationship with both the existing OpenTelemetry Schema v1.1 and the
+OpenTelemetry Semantic Conventions.
+
+## Proposal
+
+### Overview
+
+Conceptually, this proposal is based on three main concepts: **Component Telemetry
+Schema**, **Semantic Convention Registry**, and **Resolved Telemetry Schema**.
+The relationships between these entities are described in the following diagram.
+
+![Telemetry Schema Concepts](./img/0240-otel-weaver-concepts.svg)
+
+The Component Telemetry Schemas are created by the OpenTelemetry SIG members,
+application, or library authors. A Component Telemetry Schema may import any
+number of Semantic Convention
+Registries as needed. During the schema resolution process, a Resolved Telemetry
+Schema is created from a Component Telemetry Schema. This **Resolved Telemetry
+Schema is self-contained** and has no external references. Optionally, a Component
+Telemetry Schema may extend an existing Telemetry Schema, whether component or
+resolved (this aspect is still under discussion). Typically, the official
+OpenTelemetry Telemetry Schema is inherited by a Component Telemetry Schema to
+include the standard OpenTelemetry Semantic Convention registry. In complex
+cases, large enterprises might create their own intermediary telemetry schemas
+for custom definitions.
+
+Each signal definition defined in the Component Telemetry Schema, where possible,
+reuses the existing syntax and semantics defined by the semantic conventions.
+Each signal definition is also identified by a unique name (or ID), making
+schemas addressable, easy to traverse, validate, and diff.
+
+The key design principles to be followed in the definition of the Resolved
+Telemetry Schema are:
+
+* **Self-contained**: No external references are allowed. This artifact contains
+  everything required to determine what an application or a library produces in
+  terms of telemetry.
+* **Easy to exchange**: This artifact must be easily accessible from a web
+  server via a URL. This artifact must be small and avoid the repetition of
```

Comment:
```suggestion
* **Easy to exchange**: This artifact must be easily accessible to actors
  via a URI. This artifact must be small and avoid the repetition of
```

Reply:
"web server" is too specific, doesn't need to be used (e.g. maybe schemas can be loaded from a jar)